internal/resources/aws/ec2_transit_gateway_vpc_attachment.go

```diff
@@ -0,0 +1,84 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+
+	"github.com/shopspring/decimal"
+)
+
+type Ec2TransitGatewayVpcAttachment struct {
+	Address                string
+	Region                 string
+	VPCRegion              string
+	TransitGatewayRegion   string
+	MonthlyDataProcessedGB *float64 `infracost_usage:"monthly_data_processed_gb"`
+}
+
+var Ec2TransitGatewayVpcAttachmentUsageSchema = []*schema.UsageItem{{Key: "monthly_data_processed_gb", ValueType: schema.Float64, DefaultValue: 0}}
+
+func (r *Ec2TransitGatewayVpcAttachment) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *Ec2TransitGatewayVpcAttachment) BuildResource() *schema.Resource {
+	region := r.Region
+
+	if r.VPCRegion != "" {
+		region = r.VPCRegion
+	}
+
+	if r.TransitGatewayRegion != "" {
+		region = r.TransitGatewayRegion
+	}
+
+	var gbDataProcessed *decimal.Decimal
+
+	if r.MonthlyDataProcessedGB != nil {
+		gbDataProcessed = decimalPtr(decimal.NewFromFloat(*r.MonthlyDataProcessedGB))
+	}
+
+	return &schema.Resource{
+		Name: r.Address,
+		CostComponents: []*schema.CostComponent{
+			transitGatewayAttachmentCostComponent(region, "TransitGatewayVPC"),
+			transitGatewayDataProcessingCostComponent(region, "TransitGatewayVPC", gbDataProcessed),
+		}, UsageSchema: Ec2TransitGatewayVpcAttachmentUsageSchema,
+	}
+}
+
+func transitGatewayAttachmentCostComponent(region string, operation string) *schema.CostComponent {
```

Comment:
Normally I'd recommend making these methods on the struct, but it might be more difficult for these ones, so I think this works okay. What do you think?

Reply:
This method is used by different resources so I think leaving it as is is ok.